cmd/nomos/status/client.go

```diff
@@ -354,6 +361,34 @@ func (c *ClusterClient) namespaceRepoClusterStatus(ctx context.Context, cs *Clus
 	}
 }
 
+// IsOssInstallation will check for the existence of ConfigManagement object, Operator deployment, and RootSync CRD
+// If RootSync CRD exist but ConfigManagement and Operator doesn't, it indicates an OSS installation
+func (c *ClusterClient) IsOssInstallation(ctx context.Context, cs *ClusterState) bool {
+	v, cmErr := c.ConfigManagement.Version(ctx)
+	_, operatorDepErr := c.K8sClient.AppsV1().Deployments(configmanagement.ControllerNamespace).Get(ctx, ACMOperatorDeployment, metav1.GetOptions{})
+
+	if cmErr != nil || (operatorDepErr != nil && !apierrors.IsNotFound(operatorDepErr)) {
+		cs.Error = "Failed to check for operator installation status"
+		return false
```

Comment:
Here you return `false`, but the caller will continue executing, which is not necessary because you've already hit some unexpected error.
I would suggest update the `IsOssInstallation` function to return a boolean and an error.
If the error is nil, the caller can return directly.

Reply:
Updated isOss to return bool and error, updated the return condition